website/hw/hw4.rst

```diff
@@ -180,103 +218,517 @@ As you are stepping through your program, you might want to be able to
+   We have to pick a number that is unique within the machine we are running
+   on; this means that if you are running this on one of the CS Linux server,
+   and multiple people choose 50000, only one of them will actually be able
+   to run the debugger.
+
+Now, on the second terminal, run this::
+
+    gdb build/micro
+
+This will open the usual GDB prompt but, instead of typing "run", we need to
+connect to the editor that is running on the other terminal. You need
+to run the following command from the GDB prompt::
+
+    target remote :50000
+
+You will see something like this::
+
+    (gdb) target remote :50000
+    Remote debugging using :50000
+    Reading /lib64/ld-linux-x86-64.so.2 from remote target...
+    warning: File transfers from remote targets can be slow. Use "set sysroot" to access files locally instead.
+    Reading /lib64/ld-linux-x86-64.so.2 from remote target...
+    Reading symbols from target:/lib64/ld-linux-x86-64.so.2...
+    Reading /lib64/ld-2.31.so from remote target...
+    Reading /lib64/.debug/ld-2.31.so from remote target...
+    Reading /usr/lib/debug//lib64/ld-2.31.so from remote target...
+    Reading /usr/lib/debug/lib64//ld-2.31.so from remote target...
+    Reading target:/usr/lib/debug/lib64//ld-2.31.so from remote target...
+    (No debugging symbols found in target:/lib64/ld-linux-x86-64.so.2)
+    0x00007ffff7fd0100 in ?? () from target:/lib64/ld-linux-x86-64.so.2
+
+You can ignore the warning about file transfers being slow, or the message about
+"No debugging symbols" (this refers to a system library, not to our editor)
+
+You can now run any of the commands we've seen previously, with one caveat: once
+you want to start running the editor, you must use the ``continue`` command,
+and not the ``run`` command. In fact, try just running the ``continue`` command;
+GDB will show something like this::
+
+    (gdb) continue
+    Continuing.
+    Reading /lib/x86_64-linux-gnu/libc.so.6 from remote target...
+    Reading /lib/x86_64-linux-gnu/libc-2.31.so from remote target...
+    Reading /lib/x86_64-linux-gnu/.debug/libc-2.31.so from remote target...
+    Reading /usr/lib/debug//lib/x86_64-linux-gnu/libc-2.31.so from remote target...
+    Reading /usr/lib/debug//lib/x86_64-linux-gnu/libc-2.31.so from remote target...
+
+And, on the first terminal, the editor will start running.
+
+If you need to quit the editor, we suggest doing so from the first terminal
+(by pressing Ctrl-Q) but you can also force the editor to terminate the
+program by pressing Ctrl-C in GDB, followed by Ctrl-D. Please note that this
+may leave the first terminal in an unstable state (in the same way as described
+above in the case of the editor crashing). If that happens, simply perform
+the steps described earlier.
+
+
+Task 2: Stepping Through the Micro Editor Code
+----------------------------------------------
+
+(40 points)
+
+As you can see in the ``micro/main.c`` file, the ``main()`` function of our
```

Comment:
On the version I'm using it's `micro/src/main.c`; is theirs different?

Reply:
No, that's a typo. Should be fixed now.